src/main/java/uk/gov/hmcts/reform/managecase/service/NoticeOfChangeService.java

```diff
@@ -0,0 +1,207 @@
+    }
+
+    private boolean isRoleInOrganisationPolicies(List<OrganisationPolicy> organisationPolicies, String role) {
+        return organisationPolicies.stream()
+            .anyMatch(organisationPolicy -> organisationPolicy.getOrgPolicyCaseAssignedRole().equals(role));
+    }
+
+    private void validateUserRoles(CaseViewResource caseViewResource, UserInfo userInfo) {
+        List<String> roles = userInfo.getRoles();
+        if (!roles.contains(PUI_ROLE)
+            && !isActingAsSolicitor(roles, caseViewResource.getCaseType().getJurisdiction().getId())) {
+            throw new ValidationException(INSUFFICIENT_PRIVILEGE);
+        }
+    }
+
+    private boolean isActingAsSolicitor(List<String> roles, String jurisdiction) {
+        return securityUtils.hasSolicitorRole(roles, jurisdiction);
+    }
+
+    private UserInfo getUserInfo() {
+        return securityUtils.getUserInfo();
+    }
+
+    private CaseViewResource getCase(String caseId) {
+        CaseViewResource caseViewResource = new CaseViewResource();
+        try {
+            caseViewResource = dataStoreRepository.findCaseByCaseId(caseId);
+        } catch (RestClientResponseException e) {
+            if (HttpStatus.NOT_FOUND.value() == e.getRawStatusCode()) {
+                throw new CaseCouldNotBeFetchedException("Case could not be found");
+            }
+        }
+        return caseViewResource;
+    }
+
+    private CaseSearchResultViewResource findCaseBy(String caseTypeId, String caseId) {
+        return dataStoreRepository.findCaseBy(caseTypeId, null, caseId);
+    }
+
+    private void checkCaseFields(CaseSearchResultViewResource caseDetails) {
+        if (caseDetails.getCases().isEmpty()) {
+            throw new CaseCouldNotBeFetchedException("Case could not be found");
+        }
+        SearchResultViewItem searchResultViewItem = caseDetails.getCases().get(0);
+        Map<String, JsonNode> caseFields = searchResultViewItem.getFields();
+        List<SearchResultViewHeader> searchResultViewHeaderList = new ArrayList<>();
+        Optional<SearchResultViewHeaderGroup> searchResultViewHeaderGroup =
+            caseDetails.getHeaders().stream().findFirst();
+        if (searchResultViewHeaderGroup.isPresent()) {
+            searchResultViewHeaderList = searchResultViewHeaderGroup.get().getFields();
+        }
+        List<SearchResultViewHeader> filteredSearch =
+            searchResultViewHeaderList.stream()
+                .filter(searchResultViewHeader ->
+                            searchResultViewHeader.getCaseFieldTypeDefinition().getId()
+                                .equals(CHANGE_ORG_REQUEST)).collect(Collectors.toList());
+        for (SearchResultViewHeader searchResultViewHeader : filteredSearch) {
+            if (caseFields.containsKey(searchResultViewHeader.getCaseFieldId())) {
+                JsonNode node = caseFields.get(searchResultViewHeader.getCaseFieldId());
+                if (node.findValues(CASE_ROLE_ID) != null) {
```

Comment:
Can we just use `get()` here as we know what the field is meant to be called? And think we need to check it's not a `NullNode` too as seems to throw the error when the field is present with a `null` value too.

Reply:
done